Compliance Operator v0.1.57 release notes and enhancements added (#50778)
openshift/openshift-docs: modules/compliance-increasing-operator-limits.adoc

// Module included in the following assemblies:
//
// * security/compliance_operator/compliance-operator-troubleshooting.adoc

:_content-type: PROCEDURE
[id="compliance-increasing-operator-limits"]
= Increasing Compliance Operator resource limits

In some cases, the Compliance Operator might require more memory than the default limits allow. The best way to mitigate this issue is to set custom resource limits.

> To increase the default memory and CPU limits of scanner pods, see xref:../../security/compliance_operator/compliance-scans.adoc#compliance-scansetting-cr[`ScanSetting` Custom resource].

Reviewer: Same thing with the `xref`

// Module included in the following assemblies:
//
// * security/compliance_operator/compliance-operator-troubleshooting.adoc

:_content-type: PROCEDURE
[id="compliance-increasing-operator-limits"]
= Increasing Compliance Operator resource limits

In some cases, the Compliance Operator might require more memory than the default limits allow. The best way to mitigate this issue is to set custom resource limits.

To increase the default memory and CPU limits of scanner pods, see _`ScanSetting` Custom resource_.

.Procedure

> . To increase the Operator's memory limits to 500Mi, create the following patch file:

Reviewer 2: \`500Mi` or 500 Mi

Check out the "Units of measurement" section of the ISG.

I see a few other places in this PR with this issue.

+
[source,yaml]
----
spec:
  config:
    resources:
      limits:
        memory: 500Mi
----

. Apply the patch file:
+
[source,terminal]
----

> $ oc patch sub compliance-operator -nopenshift-compliance --patch-file co-memlimit-patch.yaml --type=merge

Reviewer 2: You don't specify the file name in the previous step, so it would be clearer to use a variable for the file name here. Consider making that change at some point.

----

> == Resources

Reviewer 2: Ideally, this content would be in a separate reference module.

> The `resources` field defines Resource Constraints for all the containers in the Pod created by OLM.

Reviewer 2: You should expand the first use of OLM. And it looks like "pod" shouldn't be capitalized [here](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references).

:_content-type: PROCEDURE
[id="compliance-increasing-operator-limits"]
= Increasing Compliance Operator resource limits

> In some cases, the Compliance Operator might require more memory than the default limits allow. If the Operator has been installed through OLM, the best way to mitigate issues is to set custom resource limits.

Reviewer 3: ```suggestion
If the Operator has been installed through OLM the, Compliance Operator may require more memory than the default limits allow. The best way to mitigate this particular issue is to set custom resource limits.
```

. To increase the Operator's memory limits to 500Mi, create the following patch file:
+
[source,yaml]
----
spec:
  config:
    resources:
      limits:
        memory: 500Mi
----

. Apply the patch file:
+
[source,terminal]
----
$ oc patch sub compliance-operator -nopenshift-compliance --patch-file co-memlimit-patch.yaml --type=merge
----
+
[IMPORTANT]
====
The `nodeselector` field only sets the limit for the compliance-operator
deployment, not the pods actually performing the scan.
====

== Resources

The `resources` field defines Resource Constraints for all the containers in the Pod created by OLM.

[NOTE]
====

> Resource Constraints applied in this process will overwrite existing resource constraints.

Reviewer 3: ```suggestion
Resource Constraints applied in this process will overwrite the existing resource constraints.
```

// Module included in the following assemblies:
//
// * security/compliance_operator/compliance-operator-troubleshooting.adoc

:_content-type: PROCEDURE
[id="compliance-increasing-operator-limits"]
= Increasing Compliance Operator resource limits

> If the Operator has been installed through OLM, the Compliance Operator might require more memory than the default limits allow. The best way to mitigate this issue is to set custom resource limits.

Reviewer 4: So this is tricky. You're right that in many cases the memory issues were caused by OLM's way of installing operators that watch all namespaces by default. But the memory issues are not all caused strictly by this, so I wonder if we should rephrase the first sentence to say something more generic like "In some cases, the Compliance Operator might require more memory than the default limits allow".

. To increase the Operator's memory limits to 500Mi, create the following patch file:
+
[source,yaml]
----
spec:
  config:
    resources:
      limits:
        memory: 500Mi
----

. Apply the patch file:
+
[source,terminal]
----
$ oc patch sub compliance-operator -nopenshift-compliance --patch-file co-memlimit-patch.yaml --type=merge
----
+
[IMPORTANT]
====

> The `nodeselector` field only sets the limit for the compliance-operator

Reviewer 4: nodeselector?

Reviewer 4: in addition could we cross link from here to [modules/compliance-scansetting-cr.adoc](https://github.com/openshift/openshift-docs/pull/50778/commits/10394566780a77840128b676759ae92e7c3ce349#diff-075c93655f955ab40ddb30fba38650fd721c60b7e0f3ac04447dafba58600a1b)
we already have a link from there to here, which is awesome

deployment, not the pods actually performing the scan.
====

== Resources

The `resources` field defines Resource Constraints for all the containers in the Pod created by OLM.

[NOTE]
====
Resource Constraints applied in this process will overwrite the existing resource constraints.
====

* Inject a request of 0.25 cpu and 64 MiB of memory, and a limit of 0.5 cpu and 128MiB of memory in each container by editing the `Subscription` object:
+
[source,yaml]
----
kind: Subscription
metadata:
  name: custom-operator
spec:
  package: etcd
  channel: alpha
  config:
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "500m"
----

== NodeSelector

> * The `nodeSelector` field defines a NodeSelector for the Pod created by OLM. Inject nodeSelector key-values pairs:

Reviewer 4: is there a reason to document the nodeselector here?
